pysc2/replay_parsers/base_parser.py

```diff
@@ -0,0 +1,64 @@
+# Copyright 2017 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""A base replay parser to write custom replay data scrappers."""
+
+import collections
+import six
+
+class BaseParser(object):
```

Comment:
Have you read about MapReduce? I haven't done enough research, but consider whether that would be a good framework to use here.

Reply:
Unfortunately no, not currently in my wheelhouse, but might be a good opportunity to dig into it a little.